### ISSUE TITLE
"Exclude" on declared dependencies does not work with BOM (gradle 4.6 feature)

### ISSUE THREAD

**Issue by soberich** [gradle/gradle#4567]
### Expected Behavior
After resolving bom (gradle 4.6)
the exclusion on group/module when declaring concrete dependency does exclude what is written to be excluded

### Current Behavior
Artifacts excluded from resulting war if using a full declaration with version
Artifacts ARE NOT excluded from resulting war if using a declaration without version and bom
### Context
this does NOT exclude `dom4j`
```
    implementation("org.hibernate.ogm:hibernate-ogm-bom:5.3.0.Final")
    ...
    implementation("org.hibernate:hibernate-core") {
        exclude(group: "dom4j")
    }
```
this DOES exclude `dom4j`
```
    implementation("org.hibernate:hibernate-core:5.2.13.Final") {
        exclude(group: "dom4j")
    }
```
### Your Environment

Gradle 4.6
Build time:   2018-02-28 13:36:36 UTC
Revision:     8fa6ce7945b640e6168488e4417f9bb96e4ab46c

Groovy:       2.4.12
Ant:          Apache Ant(TM) version 1.9.9 compiled on February 2 2017
JVM:          1.8.0_162 (Oracle Corporation 25.162-b12)
OS:           Windows 10 10.0 amd64

**Comment by oehme** [comment]
This is because you didn't exclude it from the dependency on the bom. I recommend declaring excludes on the configuration level. Then you don't have to worry about which dependency they might come from.

**Comment by ljacomet** [comment]
This will be fixed in Gradle 4.8 where constraints will be _only_ about the module they specific information for. Constraints will otherwise be equivalent to dependencies that have been configured as `transitive=false`.
